metaserver/src/main/scala/scala/meta/languageserver/refactoring/OrganizeImports.scala

```diff
@@ -0,0 +1,65 @@
+package scala.meta.languageserver.refactoring
+
+import scala.meta._
+import org.langmeta.internal.semanticdb._
+import org.langmeta.internal.semanticdb.schema
+import scala.meta.languageserver.Parser
+import scala.meta.languageserver.Uri
+import scala.meta.languageserver.search.SymbolIndex
+import scalafix.internal.rule.RemoveUnusedImports
+import scalafix.languageserver.ScalafixEnrichments._
+import scalafix.languageserver.ScalafixPatchEnrichments._
+import scalafix.rule.RuleCtx
+import scalafix.util.SemanticdbIndex
+import com.typesafe.scalalogging.LazyLogging
+import langserver.messages.ApplyWorkspaceEditParams
+import langserver.messages.InvalidParamsResponseError
+import langserver.types.TextDocumentIdentifier
+import langserver.types.WorkspaceEdit
+import play.api.libs.json.JsValue
+import play.api.libs.json.Json
+
+object OrganizeImports extends LazyLogging {
+
+  val empty = ApplyWorkspaceEditParams(None, WorkspaceEdit(Map.empty))
+
+  def removeUnused(
+      arguments: Option[Seq[JsValue]],
+      index: SymbolIndex
+  ): ApplyWorkspaceEditParams = {
+    val result = for {
+      as <- arguments
+      argument <- as.headOption
+      textDocument <- Json.fromJson[TextDocumentIdentifier](argument).asOpt
+    } yield removeUnused(Uri(textDocument), index)
+    result.getOrElse(
+      throw InvalidParamsResponseError(
+        s"Unable to parse TextDocumentIdentifier from $arguments"
+      )
+    )
+  }
+
+  def removeUnused(uri: Uri, index: SymbolIndex): ApplyWorkspaceEditParams = {
+    index.documentIndex.getDocument(uri) match {
+      case Some(document) =>
+        removeUnused(
+          uri,
+          schema.Database(document :: Nil).toDb(None).documents.head
```

Comment:
This reads to me like it is creating a single document database just to convert it to a different database format to then extract the single document provided initially?   Is this correct?   If so a method to name the purpose would be helpful, or just an extension method, toOtherDocumentType.

Reply:
Also, while this is not new code introduced here schema.Database(...).toDb... is also confusing.  It reads as “Create a db then convert it to a db”.